lib/ruby_lsp/executor.rb

```diff
@@ -257,6 +259,15 @@ def semantic_tokens_range(uri, range)
       )
     end
 
+    sig do
+      params(uri: String, position: Document::PositionShape).returns(T.nilable(T::Array[Interface::CompletionItem]))
+    end
+    def completion(uri, position)
+      @store.cache_fetch(uri, :completion) do |document|
```

Comment:
Since this request is position dependent, I don't think we should be caching the results.